src/current/v24.3/troubleshoot-lock-contention.md

```diff
@@ -0,0 +1,395 @@
+
+### Initial Data for Example 2
+
+In any of the SQL shells, create a second table and insert some data:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+DROP TABLE IF EXISTS t2;
+CREATE TABLE t2 (k INT PRIMARY KEY, v INT);
+INSERT INTO t2 VALUES (4,4), (5,5), (6,6);
+~~~
+
+### Example 2
+
+In this example, Transaction 4 is a write that does not block either Transaction 5, a read, or Transaction 6, a write. Transaction 4 locks key `k=4`. When Transaction 5 tries to read key `k=4`, it does not experience lock contention because it does not have to wait for the lock on the key to be released. Transaction 5 uses [`AS OF SYSTEM TIME`]({% link {{ page.version.version }}/as-of-system-time.md %}) to do a historical read. When Transaction 6 executes the [`SELECT ... FOR UPDATE NOWAIT`]({% link {{ page.version.version }}/select-for-update.md %}#wait-policies) on key `k=4`, an error is returned since the key `k=4` cannot be locked immediately. In other words, Transaction 6 "fails fast". It does not even attempt to do an `UPDATE` write to key `k=4`, so it does not experience lock contention.
+
+Transaction 4 (blocking write)   | Transaction 5 (historical read) | Transaction 6 (fail fast write)
+---------------------------------|------------------------------|--------------------------------
+`BEGIN;`                         |                              |
+`UPDATE t2 SET v=4014 WHERE k=4;`| `BEGIN AS OF SYSTEM TIME '-30s';`|
+lock k=4                         | `SELECT * FROM t2 WHERE k=4;`| `BEGIN;`
+                                 | not waiting                  | `SELECT * FROM t2 WHERE k=4 FOR UPDATE NOWAIT;`
+                                 | k=4,v=4                      | error: could not obtain lock
+`COMMIT;`                        |                              | `UPDATE t2 SET v=4034 WHERE k=4;`
+success, k=4,v=4014              |                              | not waiting
+                                 | `COMMIT;`                    | `COMMIT;`
+                                 | success                      | failure                                 
+                                 |                              | `SELECT * FROM t2 WHERE k=4;`
+                                 |                              | k=4, v=4014
+
+### SQL statements for Example 2
+
+To reproduce Example 2 in CockroachDB to show how to remediate a waiting read and write experiencing lock contention, execute the following SQL statements in the given order in the specified terminal.
```

Comment:
suggest shortening this a bit to something like

> To reproduce [Example 2](LINK), execute the following SQL statements in the given order in the specified terminal.

Reply:
shortened sentence